### PR TITLE
mgr/cephadm: Fix grafana yaml

### PR DESCRIPTION
**Pull request by sebastian-philipp**
extracted form #42378

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Review by mgfritch — APPROVED** [review]
one small typo in the commit msg: `pyhton-common` -> `python-common`

File: src/python-common/ceph/deployment/service_spec.py [hunk 1]
```diff
@@ -595,7 +595,8 @@ def to_json(self):
         if self.service_id:
             ret['service_id'] = self.service_id
         ret['service_name'] = self.service_name()
-        ret['placement'] = self.placement.to_json()
+        if self.placement.to_json():
+            ret['placement'] = self.placement.to_json()
```
Comment (sebastian-philipp) [inline src/python-common/ceph/deployment/service_spec.py:599]: > 2021-08-23T12:54:16.481 INFO:teuthology.orchestra.run.smithi061.stderr:Error EINVAL: Failed to validate Drive Group: OSD spec needs a `placement` key.

File: src/python-common/ceph/tests/test_service_spec.py [hunk 2]
```diff
@@ -233,14 +235,105 @@ def test_yaml():
...
+service_type: nfs
+service_id: mynfs
+service_name: nfs.mynfs
+spec:
+  port: 1234
+---
+service_type: iscsi
+service_id: iscsi
+service_name: iscsi.iscsi
+networks:
```
Comment (mgfritch) [inline src/python-common/ceph/tests/test_service_spec.py:275]:
```suggestion
service_name: iscsi.iscsi
placement: {}
networks:
```
test wants an empty placement dir to compare with the yaml.dump() output, but it's valid to omit the placement spec when generating a service spec....

maybe yaml.dump() should omit the placement key when empty?

**Comment by tchaikov** [comment]
jenkins test make check

**Comment by tchaikov** [comment]
- https://pulpito.ceph.com/kchai-2021-09-04_01:42:42-rados-wip-kefu-testing-2021-09-04-0729-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2021-09-04_14:03:36-rados-wip-kefu-testing-2021-09-04-0729-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2021-09-05_14:04:22-rados-wip-kefu-testing-2021-09-04-0729-distro-basic-smithi/

File: src/python-common/ceph/tests/test_service_spec.py (hunk 2 above)
Comment (tchaikov) [reply src/python-common/ceph/tests/test_service_spec.py:275]:
```
>       assert yaml.dump(object) == y
E       AssertionError: assert 'service_type...  foo: /foo\n' == 'service_type...  foo: /foo\n'
E         Skipping 74 identical leading characters in diff, use -v to show
E           llo-world
E         + placement: {}
E           spec:
E             args:
E             - --foo
E             bind_mounts:...
E         
E         ...Full output truncated (26 lines hidden), use '-vv' to show

ceph/tests/test_service_spec.py:321: AssertionError
```

@sebastian-philipp could you please consider Michael's suggestion?

Comment (sebastian-philipp) [reply src/python-common/ceph/tests/test_service_spec.py:275]: I did in #42905 but I need to rethink this more it seems

**Comment by sebastian-philipp** [comment]
already in master

**Pull request closed without merging**